src/global_scheduler/global_scheduler.h

```diff
@@ -11,11 +11,15 @@
 /* The frequency with which the global scheduler checks if there are any tasks
  * that haven't been scheduled yet. */
 #define GLOBAL_SCHEDULER_TASK_CLEANUP_MILLISECONDS 100
+#define GLOBAL_SCHEDULER_HEARTBEAT_TIMEOUT 5
```

Comment:
Can we add a comment saying that `5` refers to the number of missed heartbeats in a row that we're willing to tolerate?

Btw, I suspect this is much too low. This corresponds to a half second delay, and I wouldn't be surprised if that happens frequently.